From 878c0b0ed6a31de21c3d45d60bf5e84b0a04fccf Mon Sep 17 00:00:00 2001
From: Tomas Heinrich <theinric@redhat.com>
Date: Tue, 3 Jun 2014 12:24:53 +0200
Subject: [PATCH 3/4] Free original pointer if realloc fails

---
 src/offers.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/src/offers.c b/src/offers.c
index 7db6eee..ae34d81 100644
--- a/src/offers.c
+++ b/src/offers.c
@@ -326,20 +326,26 @@ relpOffersToString(relpOffers_t *pThis, unsigned char *pszHdr, size_t lenHdr,
 	for(pOffer = pThis->pRoot ; pOffer != NULL ; pOffer = pOffer->pNext) {
 		/* we use -3 in the realloc-guard ifs so that we have space for constants following! */
 		if(currSize - iStr - 3 < strlen((char*)pOffer->szName)) {
-			if((pszOffers = realloc(pszOffers, currSize + iAlloc)) == NULL) {
+			unsigned char *tmp;
+
+			if((tmp = realloc(pszOffers, currSize + iAlloc)) == NULL) {
 				ABORT_FINALIZE(RELP_RET_OUT_OF_MEMORY);
 				currSize += iAlloc;
 			}
+			pszOffers = tmp;
 		}
 		strcpy((char*)pszOffers+iStr, (char*)pOffer->szName);
 		iStr += strlen((char*)pOffer->szName);
 		pszOffers[iStr++] = '=';
 		for(pOfferVal = pOffer->pValueRoot ; pOfferVal != NULL ; pOfferVal = pOfferVal->pNext) {
 			if(currSize - iStr - 3 < strlen((char*)pOfferVal->szVal)) {
-				if((pszOffers = realloc(pszOffers, currSize + iAlloc)) == NULL) {
+				unsigned char *tmp;
+
+				if((tmp = realloc(pszOffers, currSize + iAlloc)) == NULL) {
 					ABORT_FINALIZE(RELP_RET_OUT_OF_MEMORY);
 					currSize += iAlloc;
 				}
+				pszOffers = tmp;
 			}
 			strcpy((char*)pszOffers+iStr, (char*)pOfferVal->szVal);
 			iStr += strlen((char*)pOfferVal->szVal);
-- 
1.9.0

