From 1f77f5c899e3fa8851e67cdc948417f2b02244df Mon Sep 17 00:00:00 2001
From: Tomas Heinrich <theinric@redhat.com>
Date: Tue, 3 Jun 2014 10:36:31 +0200
Subject: [PATCH 1/4] Correct base type for array allocation

---
 src/relp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/relp.c b/src/relp.c
index 8cfff75..459f6ff 100644
--- a/src/relp.c
+++ b/src/relp.c
@@ -644,7 +644,7 @@ engineEventLoopInit(relpEngine_t __attribute__((unused)) *pThis)
 	 */
 	for(pSrvEtry = pThis->pSrvLstRoot ; pSrvEtry != NULL ; pSrvEtry = pSrvEtry->pNext) {
 		nLstn = relpSrvGetNumLstnSocks(pSrvEtry->pSrv);
-		CHKmalloc(pSrvEtry->epevts = malloc(sizeof(epolld_t) * nLstn));
+		CHKmalloc(pSrvEtry->epevts = malloc(sizeof(epolld_t *) * nLstn));
 		for(i = 0 ; i < nLstn ; ++i) {
 			sock = relpSrvGetLstnSock(pSrvEtry->pSrv, i+1);
 			addToEpollSet(pThis, epolld_lstn, pSrvEtry->pSrv, sock, &(pSrvEtry->epevts[i]));
-- 
1.9.0

