From f3269eabe3d5cf2f06f55e4d36336a7e64932918 Mon Sep 17 00:00:00 2001
From: Tomas Heinrich <theinric@redhat.com>
Date: Tue, 3 Jun 2014 12:16:19 +0200
Subject: [PATCH 2/4] Plug mem leaks

---
 src/relp.c    | 14 ++++++++++++--
 src/relpsrv.c |  4 ++--
 2 files changed, 14 insertions(+), 4 deletions(-)

diff --git a/src/relp.c b/src/relp.c
index 459f6ff..c66432f 100644
--- a/src/relp.c
+++ b/src/relp.c
@@ -469,7 +469,8 @@ relpEngineSetOnGenericErr(relpEngine_t *pThis, void (*pCB)(char *objinfo, char*e
 relpRetVal
 relpEngineAddListner2(relpEngine_t *pThis, unsigned char *pLstnPort, void *pUsr)
 {
-	relpSrv_t *pSrv;
+	relpSrv_t *pSrv = NULL;
+
 	ENTER_RELPFUNC;
 	RELPOBJ_assert(pThis, Engine);
 	CHKRet(relpEngineListnerConstruct(pThis, &pSrv));
@@ -477,6 +478,10 @@ relpEngineAddListner2(relpEngine_t *pThis, unsigned char *pLstnPort, void *pUsr)
 	CHKRet(relpSrvSetLstnPort(pSrv, pLstnPort));
 	CHKRet(relpEngineListnerConstructFinalize(pThis, pSrv));
 finalize_it:
+	if(iRet != RELP_RET_OK) {
+		if(pSrv != NULL)
+			relpSrvDestruct(&pSrv);
+	}
 	LEAVE_RELPFUNC;
 }
 /* a dummy for callbacks not set by the caller */
@@ -506,13 +511,18 @@ relpEngineSetSyslogRcv(relpEngine_t *pThis, relpRetVal (*pCB)(unsigned char*, un
 relpRetVal
 relpEngineAddListner(relpEngine_t *pThis, unsigned char *pLstnPort)
 {
-	relpSrv_t *pSrv;
+	relpSrv_t *pSrv = NULL;
+
 	ENTER_RELPFUNC;
 	RELPOBJ_assert(pThis, Engine);
 	CHKRet(relpEngineListnerConstruct(pThis, &pSrv));
 	CHKRet(relpSrvSetLstnPort(pSrv, pLstnPort));
 	CHKRet(relpEngineListnerConstructFinalize(pThis, pSrv));
 finalize_it:
+	if(iRet != RELP_RET_OK) {
+		if(pSrv != NULL)
+			relpSrvDestruct(&pSrv);
+	}
 	LEAVE_RELPFUNC;
 }
 
diff --git a/src/relpsrv.c b/src/relpsrv.c
index 0d55346..209618b 100644
--- a/src/relpsrv.c
+++ b/src/relpsrv.c
@@ -336,7 +336,7 @@ relpSrvSetKeepAlive(relpSrv_t *pThis,
 relpRetVal
 relpSrvRun(relpSrv_t *pThis)
 {
-	relpTcp_t *pTcp;
+	relpTcp_t *pTcp = NULL;
 
 	ENTER_RELPFUNC;
 	RELPOBJ_assert(pThis, Srv);
@@ -362,7 +362,7 @@ relpSrvRun(relpSrv_t *pThis)
 
 finalize_it:
 	if(iRet != RELP_RET_OK) {
-		if(pThis->pTcp != NULL)
+		if(pTcp != NULL)
 			relpTcpDestruct(&pTcp);
 	}
 
-- 
1.9.0

