From b4c3976f3dbaa1f6b4234577a1092b3d470e49d0 Mon Sep 17 00:00:00 2001
From: Tomas Heinrich <theinric@redhat.com>
Date: Tue, 3 Jun 2014 12:54:16 +0200
Subject: [PATCH 4/4] Move misplaced code

---
 src/offers.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/offers.c b/src/offers.c
index ae34d81..252376c 100644
--- a/src/offers.c
+++ b/src/offers.c
@@ -330,9 +330,9 @@ relpOffersToString(relpOffers_t *pThis, unsigned char *pszHdr, size_t lenHdr,
 
 			if((tmp = realloc(pszOffers, currSize + iAlloc)) == NULL) {
 				ABORT_FINALIZE(RELP_RET_OUT_OF_MEMORY);
-				currSize += iAlloc;
 			}
 			pszOffers = tmp;
+			currSize += iAlloc;
 		}
 		strcpy((char*)pszOffers+iStr, (char*)pOffer->szName);
 		iStr += strlen((char*)pOffer->szName);
@@ -343,9 +343,9 @@ relpOffersToString(relpOffers_t *pThis, unsigned char *pszHdr, size_t lenHdr,
 
 				if((tmp = realloc(pszOffers, currSize + iAlloc)) == NULL) {
 					ABORT_FINALIZE(RELP_RET_OUT_OF_MEMORY);
-					currSize += iAlloc;
 				}
 				pszOffers = tmp;
+				currSize += iAlloc;
 			}
 			strcpy((char*)pszOffers+iStr, (char*)pOfferVal->szVal);
 			iStr += strlen((char*)pOfferVal->szVal);
-- 
1.9.0

